### PR TITLE
Update auditd audispd configure remote server

### PR DESCRIPTION
**Pull request by vojtapolasek**
#### Description:

- small enhancement to oval (affected families)

- add Ansible reemdiation

- remove profile metadata from tests

#### Rationale:

rhel7 stig effort

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule auditd_audispd_configure_remote_server:
 Text added outsite tags in OVAL check.
 Node deleted from OVAL check.
 Ansible remediation newly added.
 New node inserted to OVAL check.
 Node moved within OVAL check.

**Recommended tests to execute:**
 build_product fedora
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-fedora-ds.xml auditd_audispd_configure_remote_server
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-fedora-ds.xml auditd_audispd_configure_remote_server

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/ansible/shared.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+# disruption = low
+- (xccdf-var var_audispd_remote_server)
+
+- name: Make sure that a remote server is configured for Audispd
+  lineinfile:
+    {{% if product in ["rhel8", "fedora", "ol8", "rhv4"] %}}
+    path: "/etc/audit/audisp-remote.conf"
+    {{% else %}}
+    path: "/etc/audisp/audisp-remote.conf"
+    {{% endif %}}
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/ansible/shared.yml:14]: For better readability I'd carve out the Jinja `if` out of the Ansible task and define a Jinja variable.
Something similar to: https://github.com/ComplianceAsCode/content/blob/master/linux_os/guide/system/bootloader-grub2/grub2_uefi_password/oval/shared.xml#L1

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/tests/audisp_remote_server_hostname.pass.sh [hunk 2]
```diff
@@ -1,6 +1,6 @@
 #!/bin/bash
 # platform = Red Hat Enterprise Linux 7
-# profiles = xccdf_org.ssgproject.content_profile_ospp42
+
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/tests/audisp_remote_server_hostname.pass.sh:3]: Any reason for these extra empty new lines in the test scenarios?

**Comment by mildas** [comment]
**Changes identified:**
Rule auditd_audispd_configure_remote_server:
 Ansible remediation newly added.
 Text changed in OVAL check.
 Text added outsite tags in OVAL check.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.
 Node moved within OVAL check.
 Node deleted from OVAL check.
 Attribute renamed in OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml auditd_audispd_configure_remote_server
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml auditd_audispd_configure_remote_server

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/oval/shared.xml [hunk 3]
```diff
@@ -22,11 +25,7 @@
   </ind:textfilecontent54_test>
 
   <ind:textfilecontent54_object id="object_auditd_audispd_configure_remote_server" version="1">
-{{% if product in ["rhel8", "fedora", "ol8", "rhv4"] %}}
-    <ind:filepath>/etc/audit/audisp-remote.conf</ind:filepath>
-{{% else %}}
-    <ind:filepath>/etc/audisp/audisp-remote.conf</ind:filepath>
-{{% endif %}}
+    <ind:filepath>{{{ audisp_config_file_path }}}</ind:filepath>
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/oval/shared.xml:28]: I haven't noticed this, I was talking about the Ansible remediatio, :)

File: linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/oval/shared.xml [hunk 4]
```diff
@@ -1,10 +1,13 @@
...
+{{% set audisp_config_file_path = "/etc/audisp/audisp-remote.conf" %}}
+{{% endif %}}
 <def-group>
   <definition class="compliance" id="auditd_audispd_configure_remote_server" version="1">
     <metadata>
       <title>Configure audispd Plugin Remote Server IP address or Hostname</title>
-      <affected family="unix">
-        <platform>multi_platform_all</platform>
-      </affected>
+      {{{- oval_affected(products) }}}
```
Comment (yuumasato) [inline linux_os/guide/system/auditing/configure_auditd_data_retention/auditd_audispd_configure_remote_server/oval/shared.xml:10]: The description just below would also benefit from the variable: `audisp_config_file_path`

**Comment by mildas** [comment]
**Changes identified:**
Rule auditd_audispd_configure_remote_server:
 Node moved within OVAL check.
 Node deleted from OVAL check.
 Ansible remediation newly added.
 Text added outsite tags in OVAL check.
 Text changed in OVAL check.
 Deleted attribute from OVAL check.
 New node inserted to OVAL check.

**Recommended tests to execute:**
 build_product ol7
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using ansible --datastream build/ssg-ol7-ds.xml auditd_audispd_configure_remote_server
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-ol7-ds.xml auditd_audispd_configure_remote_server

**Pull request merged**